framework/resource/internal/wrapper.go

```diff
@@ -1,7 +1,36 @@
 package internal
 
-import "github.com/giantswarm/operatorkit/framework"
+import (
+	old "github.com/giantswarm/operatorkit/framework"
+	"github.com/giantswarm/operatorkit/framework/resource/internal/framework"
+)
+
+type OldWrapper interface {
+	Underlying() old.Resource
+}
+
+func OldUnderlying(r old.Resource) old.Resource {
+	for {
+		wrapper, ok := r.(OldWrapper)
+		if ok {
+			r = wrapper.Underlying()
+		} else {
+			return r
+		}
+	}
+}
```

Comment:
Those `Old*` functions are temporary for `metricresource` for the code to compile.